hikari/impl/cache.py

```diff
@@ -295,24 +357,39 @@ def _get_guilds_view(self, *, availability: bool) -> cache.CacheView[snowflakes.
         return cache_utility.CacheMappingView(results) if results else cache_utility.EmptyCacheView()
 
     def get_available_guilds_view(self) -> cache.CacheView[snowflakes.Snowflake, guilds.GatewayGuild]:
+        if not self._is_cache_enabled_for(GUILDS):
+            return cache_utility.EmptyCacheView()
+
         return self._get_guilds_view(availability=True)
 
     def get_unavailable_guilds_view(self) -> cache.CacheView[snowflakes.Snowflake, guilds.GatewayGuild]:
+        if not self._is_cache_enabled_for(GUILDS):
+            return cache_utility.EmptyCacheView()
+
         return self._get_guilds_view(availability=False)
 
     def set_guild(self, guild: guilds.GatewayGuild, /) -> None:
+        if not self._is_cache_enabled_for(GUILDS):
+            return None
+
         guild_record = self._get_or_create_guild_record(guild.id)
         guild_record.guild = copy.copy(guild)
         guild_record.is_available = True
 
     def set_guild_availability(self, guild_id: snowflakes.Snowflakeish, is_available: bool, /) -> None:
+        if not self._is_cache_enabled_for(GUILDS):
+            return None
```

Comment:
could we not methods which perform/rely on lookups implicitly pass/return None after they find nothing? Would also mean changing clear methods to do `if not resource: return empty cache view`

Reply:
Hmm, yea, could change it to that

Reply:
After thinking about this, I decided not to implement it as I prefer (and its faster) to return the default immediately if we already know its not going to be there.